### ISSUE TITLE
exception starting yajsapi during heavy load

### ISSUE THREAD

**Issue by maaktweluit** [golemfactory/yagna-triage#61]
blue - macos mojave+js+0.3.0-alpha.6 - exception while starting yajsapi blender example (happened when the machine had a loadavg of 200+)

```
    Using subnet: devnet-alpha.4, network: rinkeby, driver: zksync
    (node:47379) UnhandledPromiseRejectionWarning: TypeError: records is not iterable
        at verify_records (/Users/blue/yajsapi/examples/node_modules/yajsapi/dist/package/index.js:76:36)
        at options.callback (/Users/blue/yajsapi/examples/node_modules/srvclient/srv.js:381:13)
        at /Users/blue/yajsapi/examples/node_modules/srvclient/srv.js:350:13
        at /Users/blue/yajsapi/examples/node_modules/srvclient/srv.js:315:13
        at exports.Request.<anonymous> (/Users/blue/yajsapi/examples/node_modules/srvclient/srv.js:249:9)
        at exports.Request.emit (events.js:314:20)
        at exports.Request.Request.handleTimeout (/Users/blue/yajsapi/examples/node_modules/native-dns/lib/client.js:110:10)
        at Timeout.<anonymous> (/Users/blue/yajsapi/examples/node_modules/native-dns/lib/client.js:149:10)
        at listOnTimeout (internal/timers.js:554:17)
        at processTimers (internal/timers.js:497:7)
    (node:47379) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
    (node:47379) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by mateuszsrebrny** [comment]
@filipgolem comment:

```
this particular error happens in the resolve_repo_srv function added by @blueshade :
https://github.com/golemfactory/yajsapi/blob/8d5da0730c71b0c9eb094ad6cf4a8fe3c1c42201/yajsapi/package/index.ts#L82
maybe for (let record of records) is failing, because records is not iterable (because it is null or undefined?) and the is not iterable error is not caught.
you can add a check: if (Symbol.iterator in Object(records)) { ... } (although if (records) { ... } may be enough).

@blueshade every path must end with reject() or resolve() -> this also includes exceptions thrown e.g. if records is not iterable.

```

**Comment by filipgolem** [comment]
Fixed in `0.3.0` (commit `17a233a06244f6207dc37677d844c6a2c3352545`)
